Drivers/Periphs/Src/CANBus.c

```diff
@@ -0,0 +1,202 @@
+  * @return HAL_StatusTypeDef - Status of CAN initialization
+  */
+static HAL_StatusTypeDef MX_CAN1_Init(uint32_t mode) {
+    HAL_CAN_1->Instance = CAN1;
+    HAL_CAN_1->Init.Prescaler = 45;
+    HAL_CAN_1->Init.Mode = mode;  /* CAN_MODE_NORMAL or CAN_MODE_LOOPBACK */
+    HAL_CAN_1->Init.SyncJumpWidth = CAN_SJW_1TQ;
+    HAL_CAN_1->Init.TimeSeg1 = CAN_BS1_3TQ;
+    HAL_CAN_1->Init.TimeSeg2 = CAN_BS2_4TQ;
+    HAL_CAN_1->Init.TimeTriggeredMode = DISABLE;
+    HAL_CAN_1->Init.AutoBusOff = DISABLE;
+    HAL_CAN_1->Init.AutoWakeUp = DISABLE;
+    HAL_CAN_1->Init.AutoRetransmission = DISABLE;
+    HAL_CAN_1->Init.ReceiveFifoLocked = DISABLE;
+    HAL_CAN_1->Init.TransmitFifoPriority = DISABLE;
+
+    return HAL_CAN_Init(HAL_CAN_1);
+}
+
+
+/** CAN Config
+ * @brief Initialize CAN, configure CAN filters/interrupts, and start CAN
+ * 
+ * @param hcan pointer to CAN handle
+ * @param mode CAN_MODE_NORMAL or CAN_MODE_LOOPBACK for operation mode
+ * @param queue (QueueHandle_t) initialized FreeRTOS queue for recieved messages
+ * @return HAL_StatusTypeDef - Status of CAN configuration
+ */
+HAL_StatusTypeDef CAN_Config(
+        CAN_HandleTypeDef *hcan,
+        uint32_t mode,
+        QueueHandle_t *queue) {
+    HAL_CAN_1 = hcan;
+    RxQueue = queue;
+    HAL_StatusTypeDef configstatus = MX_CAN1_Init(mode);
+    if (configstatus != HAL_OK) return configstatus;
+
+    CAN_FilterTypeDef filterconfig;
+    filterconfig.FilterBank = FILTER_BANK;
+    filterconfig.FilterMode = CAN_FILTERMODE_IDMASK;
+    filterconfig.FilterScale = CAN_FILTERSCALE_32BIT;
+    filterconfig.FilterIdLow = FILTER_ID_LOW;
+    filterconfig.FilterIdHigh = FILTER_ID_HIGH;
+    filterconfig.FilterMaskIdLow = FILTER_MASK_ID_LOW;
+    filterconfig.FilterMaskIdHigh = FILTER_MASK_ID_HIGH;
+    filterconfig.FilterFIFOAssignment = CAN_RX_FIFO_NUMBER;
+    filterconfig.FilterActivation = CAN_FILTER_ENABLE;
+    filterconfig.SlaveStartFilterBank = 14;
+
+    // Setup filter
+    configstatus = HAL_CAN_ConfigFilter(HAL_CAN_1, &filterconfig);
+    if (configstatus != HAL_OK) return configstatus;
+
+    // Start actual CAN
+    configstatus = HAL_CAN_Start(HAL_CAN_1);
+    if (configstatus != HAL_OK) return configstatus;
+
+    // Enable interrupt for pending rx message
+    configstatus = HAL_CAN_ActivateNotification(HAL_CAN_1, 
+        (CAN_RX_FIFO_NUMBER == CAN_RX_FIFO0) ? 
+            CAN_IT_RX_FIFO0_MSG_PENDING : CAN_IT_RX_FIFO1_MSG_PENDING);
+
+    return configstatus;
+}
+
+/** CAN Fetch Message
+ * @brief Fetch a CAN message from the queue
+ * @note This is a wrapper for xQueueReceive and should not be called from an ISR
+ * 
+ * @param message Fetched message will be put here
+ * @return BaseType_t pdTRUE if CAN message was successfully fetched from queue,
+ *                    pdFALSE if queue is empty
+ */
+BaseType_t CAN_FetchMessage(CANMSG_t *message) {
+    return xQueueReceive(*RxQueue, message, (TickType_t)0);
```

Comment:
Could you reword the function name to include whether this function is blocking or non-blocking? I don't think we will need a blocking function given how many CAN messages we receive, but it might be a good idea to have one just in case.